packages/slate-editor/src/modules/editor-v4-story-bookmark/components/StoryBookmarkElement/StoryBookmarkElement.tsx

```diff
@@ -0,0 +1,85 @@
+import type { StoryBookmarkNode } from '@prezly/slate-types';
+import React, { useEffect } from 'react';
+import type { RenderElementProps } from 'slate-react';
+
+import { EditorBlock, LoadingPlaceholderV2 } from '#components';
+import { ComponentStoryBookmark } from '#icons';
+import { useAsyncFn } from '#lib';
+
+import type { StoryBookmarkExtensionParameters } from '../../types';
+
+interface Props extends RenderElementProps {
+    element: StoryBookmarkNode;
+    params: StoryBookmarkExtensionParameters;
+}
+
+const ESTIMATED_LOADING_DURATION = 300;
+
+export function StoryBookmarkElement({ attributes, children, element, params }: Props) {
+    const [{ error, loading, value: story }, loadStory] = useAsyncFn(() => {
+        return params.loadStory(element.story.uuid);
+    }, [params.loadStory, element.story.uuid]);
+
+    useEffect(() => {
+        loadStory();
+    }, [loadStory]);
+
+    return (
+        <EditorBlock
+            {...attributes} // contains `ref`
+            element={element}
+            overlay="always"
+            renderMenu={undefined}
+            renderBlock={({ isSelected }) => (
+                <div>
+                    <div>StoryBookmark</div>
+                    <pre>{JSON.stringify(element, undefined, 4)}</pre>
+                    {story && (
```

Comment:
@aspirisen could you explain that a bit for me? Why do we show plain JSON?

Reply:
It will be done in scope of other ticket, it is just a placeholder